src/phase3.hpp

```diff
@@ -399,10 +400,12 @@ Phase3Results RunPhase3(
                 Util::SliceInt64FromBytes(right_reader_entry_buf, 2 * k, right_sort_key_size);
 
             // Write the new position (index) and the sort key
-            Bits to_write = Bits(sort_key, right_sort_key_size);
-            to_write += Bits(index, k + 1);
+            uint128_t to_write = (uint128_t)sort_key << sort_key_shift;
+            to_write |= (uint128_t)index << index_shift;
 
-            L_sort_manager->AddToCache(to_write);
+            uint8_t bytes[16];
+            Util::IntTo16Bytes(bytes, to_write);
```

Comment:
this is safe without having the extra 7 bytes because it's all already aligned, right?